file.go

```diff
@@ -720,12 +809,16 @@ func initTxReleaseRegions(f *File) {
 	})
 }
 
-func withInitTx(f *File, fn func(tx *Tx) error) error {
-	tx := f.Begin()
+func withInitTx(f *File, fn func(tx *Tx) reason) reason {
+	tx, err := f.beginTx(TxOptions{Readonly: false})
```

Comment:
Seems like a change in behaviour? Looks good to me as I think it's a improvement but wanted to mention it.

Reply:
Yeah, I noticed Begin did not return an error. But it might be illegal to start a new transaction (file closed, readonly transaction). As I was improving error reporting/handling, I introduced this missing error.